src/types/utils.cpp

```diff
@@ -377,6 +1163,19 @@ til::color Utils::ColorFromHexString(const std::string_view str)
     return til::color{ r, g, b };
 }
 
+// Function Description:
+// - Parses a color from a string based on the XOrg app color name table.
+// Arguments:
+// - str: a string representation of the color name to parse
+// Return Value:
+// - A til::color if the string could successfully be parsed. If the string is not
+//      in the table, throws runtime_error
+til::color Utils::ColorFromXOrgAppColorName(const std::wstring_view wstr)
+{
+    const std::wstring key(wstr);
+    return xorgAppColorTable.at(key);

```

Comment:
how should we handle failures in this function -- should we just throw and catch an exception?

we could return a boolean and use a til::color out parameter, possibly...

Reply:
I don't honestly know. I think this particular function should only be used in OutputStateMachine once. Not like there's other places that need it. So it shouldn't really matter if it just throws. I just don't like the extra check `if (table.find(key) != table.end()` before `table.at(key)`. That's like checking twice for no good reason.

Reply:
If you cache the return value from `find` you can use it to read the mapped value directly. Check out the implementation of `at`!